contentctl/enrichments/attack_enrichment.py

```diff
@@ -25,19 +25,15 @@ def getAttackEnrichment(config:validate)->AttackEnrichment:
         _ = enrichment.get_attack_lookup(str(config.path))
         return enrichment
     
-    def getEnrichmentByMitreID(self, mitre_id:Annotated[str, Field(pattern="^T\d{4}(.\d{3})?$")])->Union[MitreAttackEnrichment,None]:
-        # TODO (cmcginley): do we want to return None when use_enrichment is false? This seems
-        #   confusing, and it seems that we intentionally do not call this function when that
-        #   setting is enabled; I'd vote to have this raise if this flag is set and this function
-        #   is called
+    def getEnrichmentByMitreID(self, mitre_id:Annotated[str, Field(pattern=r"^T\d{4}(.\d{3})?$")])->MitreAttackEnrichment:
         if not self.use_enrichment:
-            return None
+            raise Exception(f"Error, trying to add Mitre Enrichment, but use_enrichment was set to False")
```

Comment:
Changed to Exception. Agree that we should get much better with this.
As you probably know, this is a HUGE and pervasive issue in the codebase we will have to put in a lot of work to improve - but it will certainly be worth it in the long run.